### PR TITLE
add fcl/0.6.1

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **fcl/0.6.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

https://github.com/conan-io/conan-center-index/issues/621

### CODE REVIEW

**Comment by SpaceIm** [comment]
Just to mentioned two things;
- MinGW builds might fail, even with `-Wa,-mbig-obj` (in Debug mode there is little hope).
- Does self.deps_cpp_info["octomap"].version return the true version of octomap in the dependency tree (eventually overriden by consumer?)

File: recipes/fcl/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio" and self.options.shared:
```
Comment (madebr) [inline recipes/fcl/all/conanfile.py:43]: Is the problem Windows or MSVC specific? If mingw causes problems too, I would replace the test with `self.settings.os == "Windows"`

Comment (SpaceIm) [reply recipes/fcl/all/conanfile.py]: After some tests, it doesn't fail to build with MinGW and shared (at least with Release build type), but test_package fails.
I've droped support of shared on Windows, but also of Visual Studio < 2017.

**Comment by SSE4** [comment]
1. MinGW is okay to be added at later point. for now, you may just throw ConanInvalidConfiguration, or leave as is
2. yes, I think so, e.g. the same approach worked well in OpenCV recipe

**Comment by SpaceIm** [comment]
It works for MinGW as static lib, so point 1 is not longer relevant.

I've also a huge patch to fix shared lib on Windows (MinGW and Visual Studio), but honestly, I can't put that in the recipe (it exceeds size limit for one recipe). I'll submit a PR to upstream. It was hard and tedious to fix.

**Comment by madebr** [comment]
> I've also a huge patch to fix shared lib on Windows (MinGW and Visual Studio), but honestly, I can't put that in the recipe (it exceeds size limit for one recipe). I'll submit a PR to upstream. It was hard and tedious to fix.

That would be nice!
In my experience, adding a shared build to the test matrix (`.appveyor.yml`) will avoid breaking this in the future and makes upstream more willing to accept these patches.

**Comment by SpaceIm** [comment]
I've submitted a PR: https://github.com/flexible-collision-library/fcl/pull/461
As is, it will be rejected, because it bumps CMake minimum version to 3.7. Could be modified a little bit with a less elegant solution in order to preserve CMake version.

**Comment by madebr** [comment]
> I've submitted a PR: [flexible-collision-library/fcl#461](https://github.com/flexible-collision-library/fcl/pull/461)
> As is, it will be rejected, because it bumps CMake minimum version to 3.7. Could be modified a little bit with a less elegant solution in order to preserve CMake version.

Great job!
Why does it need 3.7? The reviewers are talking about supporting 3.5, which is not much less.

For similar reasons, I had to avoid using `link_options`/`target_link_options`, and use `CMAKE_EXE_LINKER_FLAGS`/`CMAKE_SHARED_LINKER_FLAGS` and similar.
I always added a FIXME about minimum required cmake version.

**Comment by SpaceIm** [comment]
Because I use an option of generate_export_header introduced in 3.7 (CUSTOM_CONTENT_FROM_VARIABLE ) in order to inject a bunch of macros (those related to explicit template instantiation and definition) at the end of export file:
https://cmake.org/cmake/help/v3.7/module/GenerateExportHeader.html

**Pull request merged**